compiler/sigmatch.nim

```diff
@@ -1702,6 +1702,15 @@ proc typeRel(c: var TCandidate, f, aOrig: PType,
               if hasCovariance:
                 continue
 
+            # Fallback for computed generic type aliases (bug #25340):
+            # Check if the aliased types (last elements) have inheritance relationship
+            let aLast = roota.last
+            let fLast = rootf.last
+            if aLast.kind == tyObject and fLast.kind == tyObject and trIsOutParam notin flags:
```

Comment:
There is no subtype relation between `G[T]` and `G[S]` just because there is one between `T` and `S`. This is not sound!